src/components/Modal.js

```diff
@@ -39,6 +39,19 @@ const Modal = (props) => {
         needsSafeAreaPadding
     } = getModalStyles(props.type, useWindowDimensions());
 
+    useEffect(() => {
+        const onKeyPressed = (event) => {
+            if (event.keyCode === 27) {
+                props.onClose();
+            }
+        };
+        document.addEventListener('keydown', onKeyPressed);
+
+        return () => {
+            document.removeEventListener('keydown', onKeyPressed);
+        };
+    }, []);
+
```

Comment:
Hey thanks! Couple of things:

1. Not sure if you caught some of the conversation in the issue, but it would be nice if we could re-use the [`KeyboardShortcut` library](https://github.com/Expensify/Expensify.cash/blob/master/src/libs/KeyboardShortcut/index.js) instead of directly referencing `document` here. One benefit to doing this is that when this code runs on native platforms it will work since we `noop` the `subscribe()` and `unsubscribe()` methods. **Reminder:** Please test your PR on all platforms.
2. We are pretty conservative when it comes to using React hooks (since we have been using class components for a long time and have not decided to switch over yet). Feel free to convert this into a class component and use lifecycle methods instead of `useEffect`.

Reply:
Okay, I just pushed two commits.

1. I used the `KeyboardShortcut` library instead of accessing the `document` directly. I slightly modified the `KeyboardShortcut` library to work with function keys.
2. I converted the `Modal` to a class component to use `componentDidMount` and `componentWillUnmount` instead of `useEffect`. In the process, I had to remove the `useWindowDimensions` hook and store the window dimensions in `state` instead.

Reply:
Perfect thanks!